File: Sem1-2/CPP_Java2024/List7/JAVA/IO.java
import java.util.NoSuchElementException;
import java.util.Scanner;

public class IO
{
    private static Scanner scanner = new Scanner(System.in);

    private IO() throws InstantiationError
    {
        throw new InstantiationError("Cannot create instance of static class IO.");
    }

    public static void setOutputLine(String line)
    {
        System.out.println(line);
    }

    public static String getInputLine()
    {
        System.out.print("> ");

        try
        {
            return scanner.nextLine().trim();
        }
        catch(NoSuchElementException e)
        {
            return null;
        }
    }

    public static void close()
    {
        scanner.close();
    }
}

File: Sem1-2/CPP_Java2024/List7/CPP/source/IO.cpp
#include <IO.hpp>

#include <iostream>

void IO::setOutputLine(const std::string& line)
{
    std::cout << line << std::endl;
}

std::string IO::getInputLine()
{
    std::cout << "> ";

    std::string line;
    std::getline(std::cin, line);
    return line;
}